src/NUnitFramework/framework/Attributes/RetryAttribute.cs

```diff
@@ -91,7 +91,15 @@ public override TestResult Execute(TestExecutionContext context)
 
                 while (count-- > 0)
                 {
-                    context.CurrentResult = innerCommand.Execute(context);
+                    try
+                    {
+                        context.CurrentResult = innerCommand.Execute(context);
+                    }
+                    catch (Exception ex)
```

Comment:
Nice... I think I'd add a comment here. Something like "All inner commands, including those created by users, are supposed to handle exceptions themselves, but in case they don't we catch it here."

Of course, some of our own commands are not handling exceptions correctly as we know. But that's for a redesign, IMO.

Reply:
I approved, so it's a suggestion. :smile:

Further suggestion would be to do it if more work is likely to wait a while rather than happening soon. I'm being cagey as I know I won't be doing it. 😈

Reply:
I'm just not looking forward to waiting for CI all over again, but that's not a good reason not to add it! 😆